main.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type bin int
+
+func (b bin) String() string {
+	return fmt.Sprintf("%b", b)
+}
+
+func main() {
+	NUMBER_RESTARTS := 100
+
+	waitingReasons := map[string]struct{}{
+		"ErrImagePull": 	{},
+		"Completed": 		{},
+		"Failed": 			{},
+		"ImagePullBackOff": {},
+	}
+
+	config, err := rest.InClusterConfig()
```

Comment:
I forgot where I saw this, but combining method call and error check into one, would make it more succinct.. 
For ex., lines 28-31 can be written as:	
```
if config, err := rest.InClusterConfig(); err != nil {
	panic(err.Error())
}
```
Same for all places where we are doing the immediate err != nil check.

Reply:
Thanks for showing me this way of writing it. I've changed it in my local.